base/client.jl

```diff
@@ -210,6 +210,14 @@ function exec_options(opts)
     history_file          = (opts.historyfile != 0)
     color_set             = (opts.color != 0) # --color!=auto
     global have_color     = color_set ? (opts.color == 1) : nothing # --color=on
+    if color_set
+        if !isa(stdout, TTY)
+            global stdout = IOContext(stdout, :color => have_color::Bool)
+        end
+        if !isa(stderr, TTY)
+            global stderr = IOContext(stderr, :color => have_color::Bool)
```

Comment:
I think fixing the issue @giordano observed https://github.com/JuliaLang/julia/pull/36689#issuecomment-659782477 requires calling `global_logger(ConsoleLogger(stderr))` again here?  This is because, at this point, `Logging.__init__` is already executed and `global_logger` is configured without `IOContext`.